core/src/main/java/org/bitcoinj/testing/FakeTxBuilder.java

```diff
@@ -73,7 +74,7 @@ public static Transaction createFakeTxWithoutChange(final NetworkParameters para
 
     /** Create a fake coinbase transaction. */
     public static Transaction createFakeCoinbaseTx(final NetworkParameters params) {
-        TransactionOutPoint outpoint = new TransactionOutPoint(params, -1, Sha256Hash.ZERO_HASH);
+        TransactionOutPoint outpoint = new TransactionOutPoint(params, ByteUtils.MAX_UNSIGNED_INTEGER, Sha256Hash.ZERO_HASH);
```

Comment:
It's fake, just like the zero hash in the next argument. It could just as well be any other value I guess.

Reply:
I see, I thought it might have special meaning at the protocol level. 

But maybe we should document `TransactionOutPoint.index` as having a maximum value. Maybe we should also consider storing it as an `int` to save space?

Reply:
The documentation is out of scope for this bugfix, it would apply to a lot of fields in the P2P protocol. One could argue that there are the BIPs to look it up.

An unsigned int doesn't really fit into a Java int – only if you never intend to do any standard math on it.